### PR TITLE
[Website] Display library assets only important files

### LINKED ISSUE #54: Do not show the whole file lists in lib page
[Bountysource](https://www.bountysource.com/issues/25535618-do-not-show-the-whole-file-lists-in-lib-page?utm_source=25535618&utm_medium=shield&utm_campaign=ISSUE_BADGE)

Some libs have too many files and the end users may not need to access them directly, should not display all of them by default, maybe hide fonts and images by default will be good, we can also hide the js/css files without minify by default, and add a button for showing the whole file list.

### PR DESCRIPTION
**Pull request by Billy4195**
Refer to #54, the library asset only display *.min.js, *.min.css, and
`filename` field in package.json.

Fixes #54, cc @PeterDaveHello.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
@Billy4195 wow, cool! Thanks!

File: webServer.js [hunk 1]
```diff
@@ -248,10 +249,14 @@ function start() {
       }
       if (assets.gennedFileNames === undefined) {
         var fileArray = [];
+        var regexpr = "{*min.js,*min.css,"+library.filename+"}";
```
Comment (PeterDaveHello) [inline webServer.js]: Can we improve the readability with better coding style here?
Some thing like: (just an example)
`var regexpr = "{*min.js,*min.css,"+library.filename+"}";`
⬇️ 
`var regexpr = "{*.min.js,*.min.css," + library.filename + "}";`

File: webServer.js [hunk 2]
```diff
@@ -248,10 +249,14 @@ function start() {
...
+        var re = globToRegExp(regexpr, { extended: true });
         assets.files.map(function(fileName, index) {
-          var fileExtension = path.extname(fileName);
-          var fileType = fileExtension.substring(1) || 'unknown';
-          fileArray.push({name: fileName, fileType: fileType});
+          if(re.test(fileName)){
```
Comment (PeterDaveHello) [inline webServer.js]: `if(re.test(fileName)){` ➡️  `if (re.test(fileName)) {`

File: webServer.js [hunk 3]
```diff
@@ -248,10 +249,14 @@ function start() {
       }
       if (assets.gennedFileNames === undefined) {
         var fileArray = [];
+        var regexpr = "{*min.js,*min.css," + library.filename + "}";
```
Comment (PeterDaveHello) [inline webServer.js]: This is not human readable enough, I think I gave the example not the answer from the last review.

To improve a little bit:

```js
"{" + "*min.js, *min.css," + library.filename + "}";
```

Comment (PeterDaveHello) [reply webServer.js]: I think I requested to use `*.min.js` instead of `*min.js` from the last review already?

Comment (Billy4195) [reply webServer.js]: @PeterDaveHello  Sorry, I forgot this.

**Comment by Billy4195** [comment]
@PeterDaveHello I added a button to show the remaining files, since the library `material-design-icons` contains too many files, click the button will let browser load files for a long long time.
Otherwise the library `JQuery` works well as following screenshots.

image
After click the button:
image

File: public/js/main.js [hunk 4]
```diff
@@ -425,4 +425,10 @@ function setFileURLs(new_provider) {
   window.onresize = function() {
     scrollProgress.update();
   };
+  $('button#show-hidden').click(function(){
+    $('tr.library').each(function(){
+        $(this).show()
```
Comment (PeterDaveHello) [inline public/js/main.js]: Wrong indent here.

**Comment by Billy4195** [comment]
@PeterDaveHello OK, I changed it.
Now if the number of asset files of a library is less than 40 files, it will display them all, otherwise it only displays files match the rule.

**Comment by PeterDaveHello** [comment]
Nice, looks better now, let's further improve the logic here, actually, the assets have different and complex patterns that we can't directly control.

So, if there are no minified css or js files with `.min.css`/`.min.css` in a library version, what should be done here?

IMO, the rule that always displays the minified files will not be suitable here, in some cases, they use `library.js`/`library.css` directly, and `library.src.js`/`library.src.css` for non-minified files.

We don't need to make the rules very complex at the beginning, we can simply display all css and js files if there is no one using `.min` the suffix in the file name, but if it's fine for you, you can prioritize the file extensions need to be hidden.

It's almost done here 👍

**Comment by Billy4195** [comment]
OK, I will do it in two weeks, because I have many projects need to be finished now.

**Comment by Billy4195** [comment]
@PeterDaveHello I have done the change.

**Comment by MattIPv4** [comment]
@Billy4195 Hey 👋 Could you possibly resolve conflicts and I can chase this to be merged?

**Comment by MattIPv4** [comment]
@PeterDaveHello Could we get this merged (I can't merge it) as I believe all issues are resolved and after testing this, I am happy that it is a good change for the site!

**Comment by PeterDaveHello** [comment]
@MattIPv4 with all due respect I'll suggest you not directly pinging me on every issues or PRs if they're just waiting for reviews, I'll always do my best to review all of them but I don't have very enough time, pinging everything may just like pinging nothing, and may even break their order in notifications, I personally prefer to handle them by the order ;) Still thanks a lot for the other part like help reviewing!

File: templates/library.html [hunk 5]
```diff
@@ -51,8 +51,13 @@ <h1 style="margin-top: 00px;" id="libraryName">{{library.name}}</h1>
       <div data-library-version="{{version}}" class="assets library-table-container">
           <table style="margin-top: 10px;" cellpadding="0" cellspacing="0" border="0" class="table table-striped" id="example">
             <tbody>
+              <tr id="control-hidden" {{^hasHidden}}style="display: none"{{/hasHidden}}>
+                <td colspan="2">
+                    <a id="show-hidden" style="cursor: pointer">Click to show all files</a>
```
Comment (PeterDaveHello) [inline templates/library.html]: I wonder if we can use wording with more information, like:
`Some files are hidden, click to show all files`

**Comment by MattIPv4** [comment]
@PeterDaveHello Just for clarification, a button to essentially toggle whether the hidden files are actually hidden?

**Comment by MattIPv4** [comment]
cc #54

**Comment by PeterDaveHello** [comment]
Correct 💯

**Pull request merged**